### PR TITLE
CI: added build with clang

### PR DESCRIPTION
**Pull request by davidbolvansky**

### CODE REVIEW

**Review by mbudiu-vmw — APPROVED** [review]

**Comment by fruffy** [comment]
This will increase our CI to 19 parallel checks. I think we should include only one build test for GMP and unified builds. I will also try to smooth out #3491 and #3485.

**Comment by davidbolvansky** [comment]
> This will increase our CI to 19 parallel checks. I think we should include only one build test for GMP and unified builds. I will also try to smooth out #3491 and #3485.

Okey, makes sense. I will do it as you proposed.

**Comment by davidbolvansky** [comment]
Done. Feel rree to merge.

**Comment by mbudiu-vmw** [comment]
@fruffy I will let you merge this

File: .github/workflows/ci-test.yml [hunk 1]
```diff
@@ -51,6 +51,35 @@ jobs:
...
+
+    - name: ccache
+      uses: hendrikmuhs/ccache-action@v1
+      with:
+        key: test-${{ runner.os }}-clang
+        max-size: 1000M
+
+    - name: Build (Ubuntu Linux, Clang)
+      run: |
+        docker build -t p4c --build-arg IMAGE_TYPE=test --build-arg ENABLE_UNIFIED_COMPILATION=ON --build-arg ENABLE_GMP=ON --build-arg ENABLE_WERROR=OFF --build-arg COMPILE_WITH_CLANG=ON .
```
Comment (fruffy) [inline .github/workflows/ci-test.yml:74]:
```suggestion
        docker build -t p4c --build-arg IMAGE_TYPE=test --build-arg ENABLE_UNIFIED_COMPILATION=ON --build-arg ENABLE_GMP=OFF --build-arg ENABLE_WERROR=OFF --build-arg COMPILE_WITH_CLANG=ON .
```

**Review by fruffy — APPROVED** [review]

**Pull request merged**